src/qt/listctrl.cpp

```diff
@@ -258,18 +1033,28 @@ bool wxListCtrl::Create(wxWindow *parent,
             const wxValidator& validator,
             const wxString& name)
 {
-    m_qtTreeWidget = new wxQtTreeWidget( parent, this );
+    m_qtTreeWidget = new wxQtListTreeWidget(parent, this);
+    m_qtTreeWidget->setModel(m_model);
 
-    if (style & wxLC_NO_HEADER)
+    if ( style & wxLC_NO_HEADER )
         m_qtTreeWidget->setHeaderHidden(true);
 
     m_qtTreeWidget->setRootIsDecorated(false);
+    m_qtTreeWidget->setSelectionBehavior(QAbstractItemView::SelectRows);
+
+    m_model->SetView(m_qtTreeWidget);
+
+    if ( !QtCreateControl(parent, id, pos, size, style, validator, name) )
+        return false;
 
-    return QtCreateControl( parent, id, pos, size, style, validator, name );
+    SetWindowStyleFlag(style);
+    return true;
 }
 
 void wxListCtrl::Init()
 {
+    m_hasCheckBoxes = false;
+    m_model = new wxQtListModel(this);
```

Comment:
Can we avoid creating the model here? It looks a pity to do it only to delete and recreate it in `Create()` for the virtual controls.